src/geo/projection/projection.js

```diff
@@ -98,6 +98,14 @@ export default class Projection {
         }
     }
 
+    isPointAboveHorizon(tr: Transform, p: Point): boolean {
+        if (tr.elevation) {
+            return !this.pointCoordinate3D(tr, p.x, p.y);
```

Comment:
nit: It might be nice to introduce a named variable so we're able to reason about that while reading it to make it explicit:

```suggestion
        if (tr.elevation) {
            const raycastOnTerrain = this.pointCoordinate3D(tr, p.x, p.y);
            return !raycastOnTerrain;
```